src/ImageSharp/Formats/Tiff/TiffDecoderCore.cs

```diff
@@ -509,6 +473,299 @@ private void DecodeStripsChunky<TPixel>(ImageFrame<TPixel> frame, int rowsPerStr
         }
     }
 
+    /// <summary>
+    /// Decodes the image data for Tiff's which arrange the pixel data in tiles and the planar configuration.
+    /// </summary>
+    /// <typeparam name="TPixel">The pixel format.</typeparam>
+    /// <param name="frame">The image frame to decode into.</param>
+    /// <param name="tileWidth">The width in pixels of the tile.</param>
+    /// <param name="tileLength">The height in pixels of the tile.</param>
+    /// <param name="tilesAcross">The number of tiles horizontally.</param>
+    /// <param name="tilesDown">The number of tiles vertically.</param>
+    /// <param name="tileOffsets">The tile offsets.</param>
+    /// <param name="tileByteCounts">The tile byte counts.</param>
+    /// <param name="cancellationToken">The token to monitor cancellation.</param>
+    private void DecodeTilesPlanar<TPixel>(
+        ImageFrame<TPixel> frame,
+        int tileWidth,
+        int tileLength,
+        int tilesAcross,
+        int tilesDown,
+        Span<ulong> tileOffsets,
+        Span<ulong> tileByteCounts,
+        CancellationToken cancellationToken)
+        where TPixel : unmanaged, IPixel<TPixel>
+    {
+        Buffer2D<TPixel> pixels = frame.PixelBuffer;
+        int width = pixels.Width;
+        int height = pixels.Height;
+        int bitsPerPixel = this.BitsPerPixel;
+        int channels = this.BitsPerSample.Channels;
+        int tilesPerChannel = tileOffsets.Length / channels;
+
+        IMemoryOwner<byte>[] tilesBuffers = new IMemoryOwner<byte>[channels];
+
+        try
+        {
+            int bytesPerTileRow = ((tileWidth * bitsPerPixel) + 7) / 8;
```

Comment:
```suggestion
            int bytesPerTileRow = (int)((((uint)tileWidth * (uint)bitsPerPixel) + 7) / 8);
```
C#: uglier
Asm: [nicer](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHunGBLAHYYGAWXIAKQcIx8ANjADqfACYYAFmgZSGwPhlwAFGFAN8EMWQEoGAXgB8DceJnylqtQwBUOvYeOnzKwYAagYAdmsAegYADgBuLh56LSFRUklUl0UVdU1tXX0jEzMLa3tHKUsnJwBXSqy3dS9HOqFLAr9iwOtQiIZomMsE6gBfIA==)

Or work with `uint` generally here (instead of `int`), then the casts aren't needed.